### PR TITLE
Add doc examples for `CString` methods.

### PR DESCRIPTION
**Pull request by frewsxcv**

### CODE REVIEW

File: src/libstd/ffi/c_str.rs [hunk 1]
```diff
@@ -288,6 +288,26 @@ impl CString {
...
+    ///
+    /// let c_string = CString::new("foo").unwrap();
+    ///
+    /// let ptr = c_string.into_raw();
+    ///
+    /// unsafe {
+    ///     assert_eq!(b'f', *ptr as u8);
+    ///     assert_eq!(b'o', *ptr.offset(1) as u8);
+    ///     assert_eq!(b'o', *ptr.offset(2) as u8);
+    ///     assert_eq!(b'\0', *ptr.offset(3) as u8);
```
Comment (QuietMisdreavus) [inline src/libstd/ffi/c_str.rs]: I'm less certain about this one. If people are used to working with raw pointers in Rust this is fine, but i wonder if we can do some other kind of comparison to not have to introduce `offset` in these docs. I don't anticipate people looking at CString with a solid understanding of Rust pointers. This is frequently an entry point into FFI, so readers may potentially only have a cursory understanding of pointers.

Comment (frewsxcv) [inline src/libstd/ffi/c_str.rs]: I'm not sure what else would fit here, let me know if you have any ideas. Or if you'd prefer the `offset` lines to be removed, I can do that too

Comment (QuietMisdreavus) [inline src/libstd/ffi/c_str.rs]: Now that I think back on it, this is probably good as-is. I was thinking of something like `slice::from_raw_parts` and comparing against a byte-string, but that risks losing focus from showing off the actual function being documented. If we come up with something better, we can always send another PR.
